src/index.js

```diff
@@ -357,6 +375,54 @@ export default class AuthnWidget {
     }
   }
 
+  registerMfaDevicePairingEventHandler() {
+    Array.from(document.querySelectorAll('[data-mfa-device-pairing-selection]'))
+      .forEach(element => element.addEventListener('click', this.handleMfaDevicePairingSelection));
+  }
+
+  handleMfaDevicePairingSelection(evt) {
+    evt.preventDefault();
+    let source = evt.currentTarget;
+    if (source) {
+      let devicePairingMethod = source.dataset['mfaDevicePairingSelection'].split('.');
+      let data = {
+        'devicePairingMethod': {
+          'deviceType': devicePairingMethod[0]
+        }
+      };
+      if (devicePairingMethod.length > 1 && devicePairingMethod[1] !== '') {
+        data['devicePairingMethod']['applicationName'] = devicePairingMethod[1];
+      }
+      this.store.dispatch('POST_FLOW', "selectDevicePairingMethod", JSON.stringify(data));
+    } else {
+      console.log("ERROR - Unable to dispatch device selection as the target was null");
+    }
+  }
+
+  async postMobileActivationRequired() {
+    let data = this.store.getStore();
+    let QRCode = require('qrcode');
+    QRCode.toCanvas(document.getElementById('qrcode'), data.pairingKey, function (error) {
+      if (error) {
+        console.error(error);
```

Comment:
Do you think its worth handling this error with something visible to the user? Can't say I know how likely this is to happen or not

Reply:
I'm not sure either -- doesn't hurt to handle it. We could also change the message "Scan the QR code using {appName} to finish pairing." to something like "Enter the pairing key using {appName} to finish pairing." instead of displaying an error?

Reply:
That seems like a nice user experience :shipit: